### PR TITLE
Check for exception in https->sendRequest()

### PR DESCRIPTION
**Pull request by pintomax**
Check for exception in https->sendRequest()

### CODE REVIEW

**Review by MonikaSharma06 — APPROVED** [review]

**Review by MonikaSharma06 — CHANGES_REQUESTED** [review]
I got following FATAL error after following the same steps -

`
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: Signal 6 (Aborted) trapped:
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (0) 0   	0x55ada06b83c9 handler(int) + 73---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (1) 1   	0x7f6bfde3df20 /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7f6bfde3df20]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (2) 2   	0x7f6bfde3de97 gsignal + 199---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (3) 3   	0x7f6bfde3f801 abort + 321---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (4) 4   	0x7f6bfe494957 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c957) [0x7f6bfe494957]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (5) 5   	0x7f6bfe49aab6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92ab6) [0x7f6bfe49aab6]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (6) 6   	0x7f6bfe49aaf1 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92af1) [0x7f6bfe49aaf1]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (7) 7   	0x7f6bfe49ad24 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92d24) [0x7f6bfe49ad24]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (8) 8   	0x7f6bfe496855 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8e855) [0x7f6bfe496855]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (9) 9   	0x7f6bef5f3e6d Hangouts::notify(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 9149---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (10) 10  	0x7f6bef5f87cb plugin_deliver + 315---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (11) 11  	0x55ada06f0c8e ./foglamp.services.notification(+0x71c8e) [0x55ada06f0c8e]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (12) 12  	0x55ada06f55f3 NotificationQueue::evalRule(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, AssetData, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, AssetData> > >&, NotificationRule*) + 1235---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (13) 13  	0x55ada06fb281 NotificationQueue::processAllDataBuffers(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 2193---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (14) 14  	0x55ada06fb487 NotificationQueue::process() + 247---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (15) 15  	0x7f6bfe4c566f /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xbd66f) [0x7f6bfe4c566f]---------
Sep  3 15:01:00 foglamp-ThinkPad-E450 FogLAMP Notifications[6527]: FATAL: (16) 16  	0x7f6bff0666db /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7f6bff0666db]---------
`

**Comment by pintomax** [comment]
@monika-Dianomic Does the Slack plugin work for you?

Is it only the google-hangouts having issues?

**Review by MonikaSharma06 — COMMENTED** [review]
Issue (mentioned in the ticket) is resolved but got another issue i.e. after removing default config of Slack Webhook URL field and Save having, error is-

`Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: Signal 6 (Aborted) trapped:
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (0) 0   	0x5645c30933c9 handler(int) + 73---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (1) 1   	0x7fb55ffa9f20 /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7fb55ffa9f20]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (2) 2   	0x7fb55ffa9e97 gsignal + 199---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (3) 3   	0x7fb55ffab801 abort + 321---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (4) 4   	0x7fb560600957 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c957) [0x7fb560600957]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (5) 5   	0x7fb560606ab6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92ab6) [0x7fb560606ab6]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (6) 6   	0x7fb560606af1 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92af1) [0x7fb560606af1]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (7) 7   	0x7fb560606d24 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92d24) [0x7fb560606d24]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (8) 8   	0x7fb560602855 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8e855) [0x7fb560602855]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (9) 9   	0x7fb55c993e4d Slack::notify(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 9149---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (10) 10  	0x7fb55c9987a8 plugin_deliver + 312---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (11) 11  	0x5645c30cbc8e ./foglamp.services.notification(+0x71c8e) [0x5645c30cbc8e]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (12) 12  	0x5645c30d05f3 NotificationQueue::evalRule(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, AssetData, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, AssetData> > >&, NotificationRule*) + 1235---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (13) 13  	0x5645c30d6281 NotificationQueue::processAllDataBuffers(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 2193---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (14) 14  	0x5645c30d6487 NotificationQueue::process() + 247---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (15) 15  	0x7fb56063166f /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xbd66f) [0x7fb56063166f]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (16) 16  	0x7fb5611d26db /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7fb5611d26db]---------
Sep  3 15:07:51 foglamp-ThinkPad-E450 FogLAMP Notifications[7101]: FATAL: (17) 17  	0x7fb56008c88f clone + 63---------`

**Comment by pintomax** [comment]
@monika-Dianomic updated with generic catch

**Review by MonikaSharma06 — APPROVED** [review]
LGTM! Now issue has been resolved.

File: hangouts.cpp [hunk 1]
```diff
@@ -63,35 +63,73 @@ SimpleHttps	*https;
 	pair<string, string> header = make_pair("Content-type", "application/json");
 	headers.push_back(header);
 
+	if (m_url.empty())
```
Comment (MonikaSharma06) [inline hangouts.cpp:66]: You should also trim it first, If I put `space` in the field, then in syslog-

`Sep  3 16:47:08 foglamp-ThinkPad-E450 FogLAMP Notifications[16987]: ERROR: Exception while sending notification to hangouts webhook  : basic_string::substr: __pos (which is 2) > this->size() (which is 1)`

Comment (pintomax) [reply hangouts.cpp:66]: Please create another JIRA

This fix for failure need to be merged to develop

**Comment by MonikaSharma06** [comment]
m_url should be checked for empty after trim. Otherwise it results in confusing error logs.

**Comment by pintomax** [comment]
FOGL-3253 just created for URL check

**Pull request merged**